test/common/markdown/markdown.tests.ps1

```diff
@@ -0,0 +1,114 @@
+$moduleRootFilePath = Split-Path -Path $PSScriptRoot -Parent
+
+# Identify the repository root path of the resource module
+$repoRootPath = (Resolve-Path -LiteralPath (Join-path $moduleRootFilePath "../..")).ProviderPath
+$repoRootPathFound = $false
+
+Describe 'Common Tests - Validate Markdown Files' -Tag 'CI' {
+    BeforeAll {
+        $skip = !$IsWindows
+        if ( !$skip ) 
+        {
+            $NpmInstalled = "not installed"
+            if (Get-Command -Name 'npm' -ErrorAction SilentlyContinue) 
+            {
+                $NpmInstalled = "Installed"
+                Write-Verbose -Message "NPM is checking Gulp is installed. This may take a few moments." -Verbose
+                $null = Start-Process `
+                    -FilePath "npm" `
+                    -ArgumentList @('install','--silent') `
+                    -Wait `
+                    -WorkingDirectory $PSScriptRoot `
+                    -PassThru `
+                    -NoNewWindow
+                $null = Start-Process `
+                    -FilePath "npm" `
+                    -ArgumentList @('install','-g','gulp','--silent') `
+                    -Wait `
+                    -WorkingDirectory $PSScriptRoot `
+                    -PassThru `
+                    -NoNewWindow
+            } 
+            elseif( -not $env:AppVeyor)
+            {
+                Write-Warning "Node and npm are required to run this test"
+                $skip = $true
+            }
+        }
+    }
+
+    AfterAll {
+        if ( !$skip ) 
+        {
+            # We're using this tool to delete the node_modules folder because it gets too long
+            # for PowerShell to remove.
+            $null = Start-Process `
+                -FilePath "npm" `
+                -ArgumentList @('install','rimraf','-g','--silent') `
+                -Wait `
+                -WorkingDirectory $PSScriptRoot `
+                -PassThru `
+                -NoNewWindow
+            $null = Start-Process `
+                -FilePath "rimraf" `
+                -ArgumentList @(Join-Path -Path $PSScriptRoot -ChildPath 'node_modules') `
```

Comment:
The whole point of this is to leave the state of your machine how it was before the test run.  We discussed this offline.